common/djangoapps/course_modes/views.py

```diff
@@ -80,18 +80,25 @@ def get(self, request, course_id, error=None):
 
         enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(request.user, course_key)
         modes = CourseMode.modes_for_course_dict(course_key)
+        ecommerce_service = EcommerceService()
 
         # We assume that, if 'professional' is one of the modes, it is the *only* mode.
         # If we offer more modes alongside 'professional' in the future, this will need to route
         # to the usual "choose your track" page same is true for no-id-professional mode.
         has_enrolled_professional = (CourseMode.is_professional_slug(enrollment_mode) and is_active)
         if CourseMode.has_professional_mode(modes) and not has_enrolled_professional:
-            return redirect(
-                reverse(
-                    'verify_student_start_flow',
-                    kwargs={'course_id': unicode(course_key)}
+            if ecommerce_service.is_enabled(request):
+                professional_mode = modes.get(CourseMode.PROFESSIONAL, '') or \
+                    modes.get(CourseMode.NO_ID_PROFESSIONAL_MODE, '')
```

Comment:
Maybe add a `CourseMode.get_professional_mode(modes)` function which you can use here instead.